shell.nim

```diff
@@ -14,6 +14,36 @@ type
     ifPipe = "|"
     ifAnd = "&&"
 
+  DebugOutputKind* = enum
+    dokCommand
+    dokError
+    dokOutput
+    dokRuntime
+
+var globalDebugConfig: set[DebugOutputKind] = {
+  dokCommand,
+  dokError,
+  dokOutput,
+  dokRuntime
+}
+
+var globalConfigChanged: bool = false
```

Comment:
I'm not a big fan of global variables like this. I understand you include it to support setting flags at runtime, but I feel like we should be able to achieve the same without globals.
Especially, because this means that all `shell` invocations will be set to the same flags. 

The `debugConfig` given to `shellVerbose` now already allows to give a variable, which can be set at runtime, no? Of course something like your example below:
```nim
  let (res, _) = shellVerbose:
￼    echo "test"
￼
￼  echo "Result is: ", res
￼
￼  setShellDebugConfig({})
￼
￼  let (res2, _) = shellVerbose:
￼    echo "test2"
```
has to be something like:
```nim
  var myConfig: set[DebugOutputKind] = someConfig()
  let (res, _) = shellVerbose(myConfig):
￼    echo "test"
￼
￼  echo "Result is: ", res
￼
￼  myConfig.clear()

￼  let (res2, _) = shellVerbose(myConfig):
￼    echo "test2"
```
But I prefer that. Global state like this can be *really* confusing. Plus in that case we don't even need the `setShellDebugConfig` and `getShellDebugConfig` procs at all. Nim's syntax for `enum` and `set` is clearer imo, too.